src/swap/utils.ts

```diff
@@ -84,26 +119,33 @@ export function getFixedOutputSwapQuote({
   assetIn: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
   assetOut: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
   amount: number | bigint;
-}): SwapQuoteWithPool {
-  const quotes = pools.map<SwapQuoteWithPool>((pool) => {
-    let quote: SwapQuote;
-    const quoteGetterArgs = {
-      pool: pool.info,
-      assetOut: {amount, id: Number(assetOut.id)},
-      decimals: {assetIn: assetIn.decimals, assetOut: assetOut.decimals},
-      reserves: pool.reserves
-    };
-
-    if (pool.info.contractVersion === CONTRACT_VERSION.V1_1) {
-      quote = SwapV1_1.getFixedOutputSwapQuote(quoteGetterArgs);
-    } else {
-      quote = SwapV2.getFixedOutputSwapQuote(quoteGetterArgs);
-    }
-
-    return {pool, quote};
+}): Promise<SwapQuoteWithPool> {
+  const quotePromises = pools.map<Promise<SwapQuoteWithPool>>((pool) => {
+    return new Promise((resolve, reject) => {
+      let quote: SwapQuote | undefined;
+
+      const quoteGetterArgs = {
+        pool: pool.info,
+        assetOut: {amount, id: Number(assetOut.id)},
+        decimals: {assetIn: assetIn.decimals, assetOut: assetOut.decimals},
+        reserves: pool.reserves
+      };
+
+      try {
+        if (pool.info.contractVersion === CONTRACT_VERSION.V1_1) {
+          quote = SwapV1_1.getFixedOutputSwapQuote(quoteGetterArgs);
+        } else {
+          quote = SwapV2.getFixedOutputSwapQuote(quoteGetterArgs);
+        }
+
+        resolve({pool, quote});
+      } catch (error) {
+        reject(error);
+      }
+    });
   });
 
-  return getBestQuote(quotes);
+  return getQuotePromiseWrapper(quotePromises);
```

Comment:
What if we do something like this instead? I think this way that new function would just be responsible on validating the quotes
```suggestion
   const validQuotes = await validateQuotes(quotePromises);

  return getBestQuote(validQuotes);
```

Reply:
Makes sense 👍